codefellowship/src/main/java/com/jencf/codefellowship/controllers/ApplicationUserController.java

```diff
@@ -131,6 +132,30 @@ public String getUserInfo(Model model, Principal principal, @PathVariable Long i
     }
 
 
+
+
+    //lab 18
+    @PutMapping("/follow-user/{id}")
+    public RedirectView followUser(Principal principal, @PathVariable Long id) {
+        ApplicationUser userToFollow = applicationUserRepo.findById(id)
+                .orElseThrow(() -> new RuntimeException("Error reading user from the database with ID of: " + id));
+
+        ApplicationUser browsingUser = applicationUserRepo.findByUsername(principal.getName());
+
+        if (browsingUser.getId().equals(userToFollow.getId())) {
+            throw new IllegalArgumentException("Following yourself is not allowed!");
+        }
+
+        browsingUser.getUsersIFollow().add(userToFollow);
+
+        applicationUserRepo.save(browsingUser);
+
+        return new RedirectView("/users/" + id);
+    }
+
+
+
+
```

Comment:
In your myprofile template, you have a link sends a Get request to "/users", but you don't have a "/users" route in your Controller.